Support multiple playbooks using results.JSONParse

When executing multiple playbooks in `playbook.AnsiblePlaybookCmd`, the JSON parsing doesn't work.

For instance, I tested with `examples/json-stdout-ansibleplaybook/json-stdout-ansibleplaybook.go`.
``` diff
--- a/examples/json-stdout-ansibleplaybook/json-stdout-ansibleplaybook.go
+++ b/examples/json-stdout-ansibleplaybook/json-stdout-ansibleplaybook.go
@@ -33,7 +33,7 @@ func main() {
        )
 
        playbook := &playbook.AnsiblePlaybookCmd{
-               Playbooks:         []string{"site.yml"},
+               Playbooks:         []string{"site.yml", "site.yml"},
                Exec:              execute,
                ConnectionOptions: ansiblePlaybookConnectionOptions,
                Options:           ansiblePlaybookOptions,
```

The result is following.
```
panic: Unmarshall error
        invalid character '{' after top-level value

goroutine 1 [running]:
main.main()
        /workspace/go-ansible/examples/json-stdout-ansibleplaybook/json-stdout-ansibleplaybook.go:50 +0x2bd
exit status 2
```

`playbook.AnsiblePlaybookCmd` uses `Playbooks` fields as an array of playbook's name, so I suggest to have an array of `AnsiblePlaybookJSONResults` as `results.JSONParse` output.

--- a/examples/json-stdout-ansibleplaybook/json-stdout-ansibleplaybook.go
+++ b/examples/json-stdout-ansibleplaybook/json-stdout-ansibleplaybook.go
@@ -13,12 +13,6 @@ import (
 )
 
 func main() {
-
-	var err error
-	var res *results.AnsiblePlaybookJSONResults
-
-	buff := new(bytes.Buffer)
-
 	ansiblePlaybookConnectionOptions := &options.AnsibleConnectionOptions{
 		Connection: "local",
 		User:       "apenella",
@@ -28,27 +22,32 @@ func main() {
 		Inventory: "127.0.0.1,",
 	}
 
+	output := io.ReadWriter(new(bytes.Buffer))
 	execute := execute.NewDefaultExecute(
-		execute.WithWrite(io.Writer(buff)),
+		execute.WithWrite(output),
 	)
 
+	playbookNames := []string{"site.yml"}
 	playbook := &playbook.AnsiblePlaybookCmd{
-		Playbooks:         []string{"site.yml"},
+		Playbooks:         playbookNames,
 		Exec:              execute,
 		ConnectionOptions: ansiblePlaybookConnectionOptions,
 		Options:           ansiblePlaybookOptions,
 		StdoutCallback:    "json",
 	}
 
-	err = playbook.Run(context.TODO())
+	err := playbook.Run(context.TODO())
 	if err != nil {
 		fmt.Println(err.Error())
 	}
 
-	res, err = results.JSONParse(buff.Bytes())
+	res, err := results.JSONParse(playbookNames, output)
 	if err != nil {
 		panic(err)
 	}
 
-	fmt.Println(res.String())
+	for _, current := range res {
+		fmt.Println(current.Playbook)
+		fmt.Println(current.String())
+	}
 }

--- a/examples/walk-through-json-output-ansibleplaybook/walk-through-json-output-ansibleplaybook.go
+++ b/examples/walk-through-json-output-ansibleplaybook/walk-through-json-output-ansibleplaybook.go
@@ -14,12 +14,6 @@ import (
 )
 
 func main() {
-
-	var err error
-	var res *results.AnsiblePlaybookJSONResults
-
-	buff := new(bytes.Buffer)
-
 	ansiblePlaybookConnectionOptions := &options.AnsibleConnectionOptions{
 		Connection: "local",
 		User:       "apenella",
@@ -29,24 +23,26 @@ func main() {
 		Inventory: "127.0.0.1,",
 	}
 
+	output := io.ReadWriter(new(bytes.Buffer))
 	execute := execute.NewDefaultExecute(
-		execute.WithWrite(io.Writer(buff)),
+		execute.WithWrite(io.Writer(output)),
 	)
 
+	playbookNames := []string{"site.yml"}
 	playbook := &playbook.AnsiblePlaybookCmd{
-		Playbooks:         []string{"site.yml"},
+		Playbooks:         playbookNames,
 		Exec:              execute,
 		ConnectionOptions: ansiblePlaybookConnectionOptions,
 		Options:           ansiblePlaybookOptions,
 		StdoutCallback:    "json",
 	}
 
-	err = playbook.Run(context.TODO())
+	err := playbook.Run(context.TODO())
 	if err != nil {
 		fmt.Println(err.Error())
 	}
 
-	res, err = results.JSONParse(buff.Bytes())
+	res, err := results.JSONParse(playbookNames, output)
 	if err != nil {
 		panic(err)
 	}
@@ -56,20 +52,22 @@ func main() {
 		Message string `json:"message"`
 	}{}
 
-	for _, play := range res.Plays {
-		for _, task := range play.Tasks {
-			if task.Task.Name == "walk-through-json-output-ansibleplaybook" {
-				for _, content := range task.Hosts {
+	for _, current := range res {
+		fmt.Println(current.Playbook)
+		for _, play := range current.Plays {
+			for _, task := range play.Tasks {
+				if task.Task.Name == "walk-through-json-output-ansibleplaybook" {
+					for _, content := range task.Hosts {
 
-					err = json.Unmarshal([]byte(content.Stdout), &msgOutput)
-					if err != nil {
-						panic(err)
+						err = json.Unmarshal([]byte(content.Stdout), &msgOutput)
+						if err != nil {
+							panic(err)
+						}
+
+						fmt.Printf("[%s] %s\n", msgOutput.Host, msgOutput.Message)
 					}
-
-					fmt.Printf("[%s] %s\n", msgOutput.Host, msgOutput.Message)
 				}
 			}
 		}
 	}
-
 }

--- a/pkg/stdoutcallback/results/JSONResults.go
+++ b/pkg/stdoutcallback/results/JSONResults.go
@@ -12,6 +12,7 @@ import (
 
 // AnsiblePlaybookJSONResults
 type AnsiblePlaybookJSONResults struct {
+	Playbook          string                                      `json:"-"`
 	CustomStats       interface{}                                 `json:"custom_stats"`
 	GlobalCustomStats interface{}                                 `json:"global_custom_stats"`
 	Plays             []AnsiblePlaybookJSONResultsPlay            `json:"plays"`
@@ -157,15 +158,20 @@ func JSONStdoutCallbackResults(ctx context.Context, r io.Reader, w io.Writer, tr
 	return nil
 }
 
-// JSONParse return an AnsiblePlaybookJSONResults from
-func JSONParse(data []byte) (*AnsiblePlaybookJSONResults, error) {
-
-	result := &AnsiblePlaybookJSONResults{}
-
-	err := json.Unmarshal(data, result)
-	if err != nil {
-		return nil, errors.New("(results::JSONParser)", "Unmarshall error", err)
+// JSONParse return an []AnsiblePlaybookJSONResults from
+func JSONParse(playbooks []string, output io.Reader) ([]AnsiblePlaybookJSONResults, error) {
+	results := make([]AnsiblePlaybookJSONResults, len(playbooks))
+	decoder := json.NewDecoder(output)
+	for index := 0; index < len(playbooks); index++ {
+		err := decoder.Decode(&results[index])
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, errors.New("(results::JSONParser)", "Unmarshall error", err)
+		}
+		results[index].Playbook = playbooks[index]
 	}
 
-	return result, nil
+	return results, nil
 }

--- a/pkg/stdoutcallback/results/JSONResults_test.go
+++ b/pkg/stdoutcallback/results/JSONResults_test.go
@@ -287,7 +287,7 @@ func TestJSONParser(t *testing.T) {
 	tests := []struct {
 		desc        string
 		inputResult string
-		res         *AnsiblePlaybookJSONResults
+		res         []AnsiblePlaybookJSONResults
 	}{
 		{
 			desc: "Testing json parse",
@@ -339,51 +339,54 @@ func TestJSONParser(t *testing.T) {
 					}
 				}
 			}`,
-			res: &AnsiblePlaybookJSONResults{
-				CustomStats:       map[string]interface{}{},
-				GlobalCustomStats: map[string]interface{}{},
-				Plays: []AnsiblePlaybookJSONResultsPlay{
-					{
-						Play: &AnsiblePlaybookJSONResultsPlaysPlay{
-							Name: "local",
-							Id:   "a0a4c5d1-62fd-b6f1-98ea-000000000006",
-							Duration: &AnsiblePlaybookJSONResultsPlayDuration{
-								End:   "2020-08-07T20:51:30.942955Z",
-								Start: "2020-08-07T20:51:30.607525Z",
-							},
-						},
-						Tasks: []AnsiblePlaybookJSONResultsPlayTask{
-							{
-								Task: &AnsiblePlaybookJSONResultsPlayTaskItem{
-									Id:   "a0a4c5d1-62fd-b6f1-98ea-000000000008",
-									Name: "Print line",
-									Duration: &AnsiblePlaybookJSONResultsPlayTaskItemDuration{
-										End:   "2020-08-07T20:51:30.942955Z",
-										Start: "2020-08-07T20:51:30.908539Z",
-									},
+			res: []AnsiblePlaybookJSONResults{
+				{
+					Playbook:          "test.yml",
+					CustomStats:       map[string]interface{}{},
+					GlobalCustomStats: map[string]interface{}{},
+					Plays: []AnsiblePlaybookJSONResultsPlay{
+						{
+							Play: &AnsiblePlaybookJSONResultsPlaysPlay{
+								Name: "local",
+								Id:   "a0a4c5d1-62fd-b6f1-98ea-000000000006",
+								Duration: &AnsiblePlaybookJSONResultsPlayDuration{
+									End:   "2020-08-07T20:51:30.942955Z",
+									Start: "2020-08-07T20:51:30.607525Z",
 								},
-								// TODOx
-								Hosts: map[string]*AnsiblePlaybookJSONResultsPlayTaskHostsItem{
-									"127.0.0.1": {
-										//"_ansible_no_log": false, "_ansible_verbose_always": true,
-										Action:  "debug",
-										Changed: false,
-										Msg:     []interface{}{"That's a message to debug"},
+							},
+							Tasks: []AnsiblePlaybookJSONResultsPlayTask{
+								{
+									Task: &AnsiblePlaybookJSONResultsPlayTaskItem{
+										Id:   "a0a4c5d1-62fd-b6f1-98ea-000000000008",
+										Name: "Print line",
+										Duration: &AnsiblePlaybookJSONResultsPlayTaskItemDuration{
+											End:   "2020-08-07T20:51:30.942955Z",
+											Start: "2020-08-07T20:51:30.908539Z",
+										},
+									},
+									// TODOx
+									Hosts: map[string]*AnsiblePlaybookJSONResultsPlayTaskHostsItem{
+										"127.0.0.1": {
+											//"_ansible_no_log": false, "_ansible_verbose_always": true,
+											Action:  "debug",
+											Changed: false,
+											Msg:     []interface{}{"That's a message to debug"},
+										},
 									},
 								},
 							},
 						},
 					},
-				},
-				Stats: map[string]*AnsiblePlaybookJSONResultsStats{
-					"127.0.0.1": {
-						Changed:     0,
-						Failures:    0,
-						Ignored:     0,
-						Ok:          1,
-						Rescued:     0,
-						Skipped:     0,
-						Unreachable: 0,
+					Stats: map[string]*AnsiblePlaybookJSONResultsStats{
+						"127.0.0.1": {
+							Changed:     0,
+							Failures:    0,
+							Ignored:     0,
+							Ok:          1,
+							Rescued:     0,
+							Skipped:     0,
+							Unreachable: 0,
+						},
 					},
 				},
 			},
@@ -392,7 +395,7 @@ func TestJSONParser(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.desc, func(t *testing.T) {
-			res, _ := JSONParse([]byte(test.inputResult))
+			res, _ := JSONParse([]string{"test.yml"}, strings.NewReader(test.inputResult))
 			assert.Equal(t, test.res, res, "Unexpected result")
 		})
 	}

